echopype/convert/parse_base.py

```diff
@@ -89,13 +89,26 @@ def rectangularize_data(self):
                     self.ch_ids[data_type].append(k)
                     self.ping_data_dict[data_type][k] = self.pad_shorter_ping(v)
             # Transmit data
-            for k, v in self.ping_data_dict_tx[data_type].items():
-                if all(
-                    (x is None) or (x.size == 0) for x in v
-                ):  # if no data in a particular channel
-                    self.ping_data_dict_tx[data_type][k] = None
-                else:
-                    self.ping_data_dict_tx[data_type][k] = self.pad_shorter_ping(v)
+            self.rectangularize_transmit_ping_data(data_type)
+
+    def rectangularize_transmit_ping_data(self, data_type: str) -> None:
+        """
+        Rectangularize the ``data_type`` data within transmit ping data.
+        Additionally, convert the data to a numpy array
+        indexed by channel.
+
+        Parameters
+        ----------
+        data_type: str
+            The key of ``self.ping_data_dict_tx`` to rectangularize
```

Comment:
the only `data_type` that are specified is "complex" so I think we can hard-code this.

Reply:
As we discussed, we will not change this for now.